src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Operations;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.InlineMethod
+{
+    internal abstract partial class AbstractInlineMethodRefactoringProvider<TInvocationNode, TExpression, TArgumentSyntax, TMethodDeclarationSyntax, TIdentifierName>
+        where TInvocationNode : SyntaxNode
+        where TExpression : SyntaxNode
+        where TArgumentSyntax : SyntaxNode
+        where TMethodDeclarationSyntax : SyntaxNode
+        where TIdentifierName : SyntaxNode
+    {
+        /// <summary>
+        /// Information about the callee method parameters to compute <see cref="InlineMethodContext"/>.
+        /// </summary>
+        private class MethodParametersInfo
+        {
+            /// <summary>
+            /// Parameters map to identifier argument. The identifier from Caller will be used to replace
```

Comment:
you say it's a map, but it's an array fo tuples.  is the array form preferred?  or would a dictionary be more suitable here?

Reply:
For this one it could be a map